cpp/src/DataStorm/NodeI.cpp

```diff
@@ -223,29 +244,54 @@ NodeI::createSession(
                         uncheckedCast<PublisherSessionPrx>(session->getProxy()),
                         nullptr,
                         [self, subscriber, session](auto ex)
-                        { self->removePublisherSession(*subscriber, session, ex); });
+                        { self->retryPublisherSessionCreation(*subscriber, session, ex); });
```

Comment:
The logic is the same but the intention is now clear. We attempt a retry, and only remove the session if a retry is not possible.